contracts/SpokePool.sol

```diff
@@ -1,16 +1,161 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.0;
 
-contract SpokePool {
-    //   constructor() {}
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@uma/core/contracts/common/implementation/Testable.sol";
+import "@uma/core/contracts/common/implementation/Lockable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
 
+/**
+ * @title SpokePool
+ * @notice Contract deployed on source and destination chains enabling depositors to transfer assets from source to 
+ * destination. Deposit orders are fulfilled by off-chain relayers who also interact with this contract. Deposited
+ * tokens are locked on the source chain and relayers send the recipient the desired token currency and amount
+ * on the destination chain. Locked source chain tokens are later sent over the canonical token bridge to L1.
+ */
+contract SpokePool is Testable, Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+
+    // Timestamp when contract was constructed. Relays cannot have a quote time before this.
+    uint64 public deploymentTime;
+
+    // Track the total number of deposits. Used as a unique identifier for deposits.
+    uint256 public numberOfDeposits;
+
+    struct DestinationToken {
+        address token;
+        address spokePool;
+        address wethContract;
+        bool depositsEnabled;
+    }
+
+    // Whitelist of origin token to destination token routings.
+    mapping(address => mapping(uint256 => DestinationToken)) public whitelistedDestinationRoutes;
+
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+    event WhitelistRoute(address originToken, uint256 destinationChainId, address destinationToken, address spokePool, address wethContract);
+    event DepositsEnabled(address originToken, uint256 destinationChainId, bool depositsEnabled);
+    event FundsDeposited(
+        uint256 nonce,
```

Comment:
should we call this something different? should it be `depositId`?

Reply:
what do you think? I'm good with either.

Reply:
ye I think there is a good meta point here: where posible (and even when not 100% posible) we should try and stick with the same interface/variable name as in v1 to simplify bot/front end compatibility.

Reply:
cool will change